crates/router/src/connector/stripe.rs

```diff
@@ -1929,3 +1952,484 @@ impl services::ConnectorRedirectResponse for Stripe {
+            customer_account.into_masked(),
+        )];
+        headers.append(&mut customer_account_header);
+        Ok(headers)
+    }
+
+    fn get_request_body(
+        &self,
+        req: &types::PayoutsRouterData<api::PoFulfill>,
+    ) -> CustomResult<Option<types::RequestBody>, errors::ConnectorError> {
+        let connector_req = stripe::StripeConnectPayoutFulfillRequest::try_from(req)?;
+        let stripe_connect_req = types::RequestBody::log_and_get_request_body(
+            &connector_req,
+            utils::Encode::<stripe::StripeConnectPayoutFulfillRequest>::url_encode,
+        )
+        .change_context(errors::ConnectorError::RequestEncodingFailed)?;
+        Ok(Some(stripe_connect_req))
+    }
+
+    fn build_request(
+        &self,
+        req: &types::PayoutsRouterData<api::PoFulfill>,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<Option<services::Request>, errors::ConnectorError> {
+        let request = services::RequestBuilder::new()
+            .method(services::Method::Post)
+            .url(&types::PayoutFulfillType::get_url(self, req, connectors)?)
+            .attach_default_headers()
+            .headers(types::PayoutFulfillType::get_headers(
+                self, req, connectors,
+            )?)
+            .body(types::PayoutFulfillType::get_request_body(self, req)?)
+            .build();
+
+        Ok(Some(request))
+    }
+
+    #[instrument(skip_all)]
+    fn handle_response(
+        &self,
+        data: &types::PayoutsRouterData<api::PoFulfill>,
+        res: types::Response,
+    ) -> CustomResult<types::PayoutsRouterData<api::PoFulfill>, errors::ConnectorError> {
+        let response: stripe::StripeConnectPayoutFulfillResponse = res
+            .response
+            .parse_struct("StripeConnectPayoutFulfillResponse")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        types::RouterData::try_from(types::ResponseRouterData {
+            response,
+            data: data.clone(),
+            http_code: res.status_code,
+        })
+    }
+
+    fn get_error_response(
+        &self,
+        res: types::Response,
+    ) -> CustomResult<types::ErrorResponse, errors::ConnectorError> {
+        self.build_error_response(res)
+    }
+}
+
+#[async_trait::async_trait]
+#[cfg(feature = "payouts")]
+impl
+    services::ConnectorIntegration<api::PoRecipient, types::PayoutsData, types::PayoutsResponseData>
+    for Stripe
+{
+    fn get_content_type(&self) -> &'static str {
+        self.common_get_content_type()
+    }
+
+    async fn execute_posttasks(
+        &self,
+        router_data: &mut types::PayoutsRouterData<api::PoRecipient>,
+        app_state: &routes::AppState,
+    ) -> CustomResult<(), errors::ConnectorError> {
+        // Adding artificial delay since account capabilities take some time to reflect in Stripe's platform
+        sleep(Duration::from_millis(5000)).await;
```

Comment:
Is this the avg/anticipated delay, If not Can it be reduced to a minimum as it would be directly impacting the application performance and latency

Reply:
This is the avg delay we have seen in Stripe's test mode so far. Will check this again and update it in the future. For now, I will move it to constants module